lib/rubygems/gemcutter_utilities.rb

```diff
@@ -296,6 +301,18 @@ def webauthn_verification_url(credentials)
     response.is_a?(Net::HTTPSuccess) ? response.body : nil
   end
 
+  def default_browser_safari?
+    return false unless RUBY_PLATFORM.include?("darwin")
+
+    require "json"
+
+    launch_services = JSON.parse(`defaults read com.apple.LaunchServices/com.apple.launchservices.secure LSHandlers \
```

Comment:
:thinking: What about to wrap this, rescue any exception raised and return false on troubles? It would not make sense to break RubyGems just because system will change those internal commands.